common/djangoapps/util/enterprise_helpers.py

```diff
@@ -0,0 +1,118 @@
+try:
+    from enterprise.models import EnterpriseCustomer
+    from enterprise.tpa_pipeline import (
+        active_provider_requests_data_sharing,
+        active_provider_enforces_data_sharing,
+        get_enterprise_customer_for_request,
+    )
+
+except ImportError:
+    pass
+
+from django.conf import settings
+
+
+def enterprise_enabled():
+    """
+    Determines whether the Enterprise app is installed
+    """
+    return 'enterprise' in settings.INSTALLED_APPS
+
+
+def data_sharing_consent_requested(request):
+    """
+    Determine if the EnterpriseCustomer for a given HTTP request
+    requests data sharing consent
+    """
+    if enterprise_enabled():
+        return active_provider_requests_data_sharing(request)
+    else:
+        return False
+
+
+def data_sharing_consent_required_at_login(request):
+    """
+    Determines if data sharing consent is required at
+    a given location
+    """
+    if enterprise_enabled():
+        return active_provider_enforces_data_sharing(request, EnterpriseCustomer.AT_LOGIN)
+    else:
+        return False
+
+
+def data_sharing_consent_requirement_at_login(request):
+    """
+    Returns either 'optional' or 'required' based on where we are.
+    """
+    if not enterprise_enabled():
+        return None
+    if data_sharing_consent_required(request, EnterpriseCustomer.AT_LOGIN):
+        return 'required'
+    if data_sharing_consent_requested(request):
+        return 'optional'
+    return None
+
+def insert_enterprise_fields(request, form_desc):
+    """
+    For right now, a simple wrapper for `add_data_sharing_consent_field`,
+    but we might want to add other form fields later.
+    """
+    if not enterprise_enabled():
+        return
+    add_data_sharing_consent_field(request, form_desc)
+
+
+def add_data_sharing_consent_field(request, form_desc):
+    """
+    Adds a checkbox field to be selected if the user consents to share data with
+    the EnterpriseCustomer attached to the SSO provider with which they're authenticating.
+    """
+    enterprise_customer = get_enterprise_customer_for_request(request)
+    required = data_sharing_consent_required_at_login()
```

Comment:
Yep, ran into that while doing manual testing. Fix in just a moment. For whatever reason, my devstack edx-platform is no longer getting mounted properly on my host, so the only effective way to test is to push and pull.